dyne/ukf.py

```diff
@@ -5,6 +5,41 @@
 from ._common import check_input_arrays, check_measurements
 
 
+def _unscented_transform(x, P, Q, alpha):
+    """Perform "unscented transform"
+
+    At first augument state distribution with mean `x` and covariance `P`
+    with zero mean noise and covariance Q. Then sample ``2 * n points`` from
+    this distribution.
+
+    Returns
+    -------
+    X_probe : ndarray, shape (2*n, n)
+        "Sigma points" i.e. samples from augumented distribution.
+    W_m : float
+        Weights of samples to calculate mean. Mean can be calculated as
+        W_m * np.sum(X_probe, axis=0).
+    W_c : float
+        Weights of samples to calculate covariance. Covariance matrix can be
+        caliculated as W_c * X_probe.T @ X_probe.
+    """
+    n_states = len(x)
+    n_noises = len(Q)
+    l, V = linalg.eigh(P)
+    # Due to numerical issues computed eigenvalues could be negative,
+    # here we set them to zero
+    l[l < 0] = 0
+    Sigma = alpha * np.sqrt(n_states + n_noises) * linalg.block_diag(
+        np.diag(l**0.5) @ V.T, linalg.cholesky(Q))
+    X_probe = []
+    for sigma in Sigma:
+        for sign in [-1, 1]:
+            X_probe.append(np.hstack((x + sign * sigma[:n_states],
+                                          sign * sigma[n_states:])))
```

Comment:
This implicit 0 is sort of awkward.

I know it works, but I think a more general function (as I hinted) will be a more elegant and clean solution.